airflow/providers/cncf/kubernetes/provider.yaml

```diff
@@ -350,6 +350,15 @@ config:
         type: string
         example: ~
         default: ""
+      task_publish_max_retries:
+        description: |
+          The Maximum number of retries for queuing the task to the kubernetes scheduler when
+          failing due to Kube API transient errors before giving up and marking task as failed.
```

Comment:
This isn't for all transient errors, this only covers quota failures right? We might want to change the name to reflect this somehow as well?

Reply:
All errors except (400, 404 and 422) will go for retry based on task_publish_max_retries configuration.

Reply:
We don't need to retry for all errors, just the exceeded quota, +1 for applying my comment above and renaming the config.

Reply:
The existing code put the task in the queue indefinitely for all the error codes except (400, 404, 422). I have retained the old behavior with a -1 flag. Do you want me to requeue the task when the quota is exceeded only, then fail the rest of the error codes?

Reply:
I have addressed the review comments.